.changeset/nasty-forks-beg.md

```diff
@@ -0,0 +1,15 @@
+---
+'@backstage/plugin-search-react': minor
+'@backstage/plugin-search': minor
+---
+
+The SearchPage component can now be configured via app-config.yaml with an initial state to define how it behaves when it is first loaded or reset. Check out the following example:
+
+```yaml
+app:
+  search:
+    initialState:
```

Comment:
What if we use query instead of initialState? This setting could potentially be universal (used not only by frontend plugins but also by backend 🎉). I see an opportunity for us to start using it in search engine implementations (for example, here on the [Elasticsearch](https://github.com/backstage/backstage/blob/9e826a3f8796b5f5bfd2f2eb31406c9bec327bdd/plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts#L239), [Postgres](https://github.com/backstage/backstage/blob/9e826a3f8796b5f5bfd2f2eb31406c9bec327bdd/plugins/search-backend-module-pg/src/PgSearchEngine/PgSearchEngine.ts#L151) and [Lurn](https://github.com/backstage/backstage/blob/9e826a3f8796b5f5bfd2f2eb31406c9bec327bdd/plugins/search-backend-node/src/engines/LunrSearchEngine.ts#L76) query translators).